### DISCUSSION TITLE
How to configure new post settings?

### DISCUSSION THREAD

**Discussion by Wysardry** [getnikola/nikola#3752] [Q&A]
I'm new to Python, so I'm having trouble working out how to change the settings in conf.py

I would like new posts and pages to be created in markdown (instead of reST) by default. I would also like additional metadata fields to be added.

I've found the ADDITIONAL_METADATA field to set the latter, but am unsure of the format to use, especially for the standard variables such as 'updated'.

Should I be using single or double quotes for variable names and values? How can I tell Nikola to set 'updated' to be the same as 'date' when a new post is created?

Finally, is it possible to change the sort order for meta data in new posts? It seems to default to alphabetical order, which means that 'date' and 'updated' aren't very close to each other.

**Comment by Kwpolska** [comment]
> I would like new posts and pages to be created in markdown (instead of reST) by default.

Make the first entry in POSTS and PAGES the one for Markdown.

> Should I be using single or double quotes for variable names and values?

Either works. The syntax for a Python dictionary is `{"key": "value"}`, you can use a variable instead of strings if you omit quotes.

> How can I tell Nikola to set 'updated' to be the same as 'date' when a new post is created?

For `updated`, I would recommend not setting a value if it’s the same as `date`. You can’t tell Nikola to use the same date. You could try getting the current date using the `datetime` module, but that will probably be off by a few seconds from the date used by `new_post`.

> Finally, is it possible to change the sort order for meta data in new posts? It seems to default to alphabetical order, which means that 'date' and 'updated' aren't very close to each other.

That’s not possible.

**Comment by Wysardry** [comment]
Thanks. Is 'updated' intended to be set manually, or does Nikola use the file's time stamp to set it?

**Reply by Kwpolska** [reply]
I checked and Nikola will actually set `updated` to be equal to `date` by default. You can set `updated` to another value if the post was updated after original publication.

**Comment by Wysardry** [comment]
So it's best to only add a value to 'updated' when the text is _actually_ updated? If I set it as '' so I can fill in the blank later, it causes several errors.

**Reply by Kwpolska** [reply]
The best thing to do is to only _add_ the `updated` meta field to the post if the post was ever updated. You may also leave it blank — this means no quotes if you're using the default format.

**Reply by Wysardry** [reply]
Considering the options, I think you're right about not adding the updated field until an actual change is made.

If it defaults to the same value as the date, then it would make sorting by the updated field easier, as I wouldn't need to check if it was blank (and sort by date instead if it was).
